lib/testing/test_runner.sh

```diff
@@ -123,7 +123,11 @@ check_prereq()
         die "This script needs root permissions to run."
     fi
 
-    [ -d "$STATEDIR" ] || mkdir -p "$STATEDIR" || die "Unable to create state dir $STATEDIR"
+    STATEDIR="$(mktemp -d --tmpdir=${TMPDIR:-/tmp} --suffix=.xdptest)" 
+    if [ $? -ne 0 ]; then
+        die "Unable to create state dir in $TMPDIR"
+    fi
+    mkdir ${STATEDIR}/proc
```

Comment:
cleanup() does `rmdir $STATEDIR`, so with this I think it'll fail, no? I think the easiest is to just change the cleanup to 'rm -rf' :)

Reply:
Looking at the code, I'm having trouble finding where the cleanup() function is even used. I thought that the cleaning was done in teardown().

Reply:
Huh, you're right! Looks like cleanup() is just dead code left over from when this was copied from the tutorial runner. My bad!